.github/workflows/master.yml

```
@@ -0,0 +1,77 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2024 Intel Corporation
+
+name: Master workflow
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+      - name: Build
+        run: |
+          go build -o build/_output/onos ./cmd/onos
+          go build -o build/_output/onos-cli-docs-gen ./cmd/onos-cli-docs-gen
+          go build -o build/_output/gnmi_cli ./cmd/gnmi_cli
+
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+      - uses: golangci/golangci-lint-action@v3.7.0
+        with:
+          version: latest
+          args: -v --config ./.golangci.yml
+
+  unit-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+      - name: Unit tests
+        run: |
+          go test github.com/onosproject/onos-cli/pkg/...
+          go test github.com/onosproject/onos-cli/cmd/...
+
+  docker-build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+      - name: Build Docker image
+        run: |
+          go mod vendor
+          docker build . -f build/onos/Dockerfile \
+                  -t onosproject/onos-cli:latest
+          rm -rf vendor
```

If `vendor` is not needed, then you can do something like this:
```suggestion
        run: docker build . -f build/onos/Dockerfile -t onosproject/onos-cli:latest
```

---

FYI - there are two reasons for using a `vendor` directory at the top level.
- copying from the host to the build docker is faster than downloading the dependencies again inside the container
- In a development environment where you want to build against local changes to another repo (`onos-api` is a common one), the build container does not have access to those files on the host, but if you use `go mod vendor` along with a `replace` directive in the `go.mod`, you will have access in both the host and the container

---

Thanks @ray-milkey for the details